source/module_hsolver/diago_david.cpp

```diff
@@ -976,8 +984,9 @@ void DiagoDavid<T, Device>::SchmitOrth(const int& dim,
 template <typename T, typename Device>
 void DiagoDavid<T, Device>::planSchmitOrth(const int nband, int* pre_matrix_mm_m, int* pre_matrix_mv_m)
```

Comment:
is it possible to replace int* with vector<int>?

Reply:
`planSchmitOrth` is modified to replace the use of raw pointer types with `std::vector<int>` references.
The initialization of `std::vector<int>` inside this function was previously handled by a custom `ModuleBase::GlobalFunc::ZEROS` function, which appears to be a simple custom memory initialization function to set elements of an array to zero.
Is it necessary to replace it with a standard `std::vector` initialization function?

Reply:
I suggest doing so.